zephyr/wrapper.c

```diff
@@ -68,9 +68,9 @@ __section(".heap_mem") static uint8_t __aligned(64) heapmem[HEAPMEM_SIZE];
 
 #else
 
-extern uint8_t _end, _heap_sentry;
-#define heapmem ((uint8_t *)ALIGN_UP((uintptr_t)&_end, PLATFORM_DCACHE_ALIGN))
-#define HEAPMEM_SIZE (&_heap_sentry - heapmem)
+extern char _end[], _heap_sentry[];
+#define heapmem ((char *)ALIGN_UP((uintptr_t)_end, PLATFORM_DCACHE_ALIGN))
```

Comment:
Why change the type of `heapmem`? This makes it inconsistent with CONFIG_IMX above.

I don't think `heapmem` must have the same type as `_end`. If you think it must, then change `heapmem` in CONFIG_IMX too.

Granted, `sys_heap_init()` accepts a `void *` (=anything) but the inconsistency with CONFIG_IMX could become an issue if `heapmem` ever gets used somewhere else.

Reply:
No need - just need to be aligned for the subtraction on `HEAPMEM_SIZE`. Just updated to cast `_heap_sentry` there instead.